soap/soap.go

```diff
@@ -228,8 +229,17 @@ func (s *Client) AddHeader(header interface{}) {
 	s.headers = append(s.headers, header)
 }
 
+// CallCtx performs HTTP POST request with a context
+func (s *Client) CallCtx(ctx context.Context, soapAction string, request, response interface{}) error {
```

Comment:
Let's keep it consistent with Go's standard library and name this `CallContext` instead.